### PR TITLE
Event G-code for Stop SD Print

### PR DESCRIPTION
**Pull request by robbycandra**
Add option to park Nozzle when Stop Print. 
Nozzle park could be needed to prevent Nozzle Heat damage the printed object

Related to issue #13517

### CODE REVIEW

**Comment by 7eggert** [comment]
Great minds think alike, but your solution is more specific.

https://github.com/MarlinFirmware/Marlin/pull/12850/commits/c9bf059428bcab6173d1de04b9010b2fa94caa9a (#12850)

**Comment by robbycandra** [comment]
@7eggert thanks....
I just think that we dont need 2 feature to do the same function. 
Because marlin already have Nozzle Park, then why not use it.

My Idea... we have 2 fixed position in the printer. 
1. Home Position -> We use this position when we start printing.
2. Nozzle Park Positition -> Nozzle goes here when stop printing.

**Comment by 7eggert** [comment]
You would inject the nozzle park feature as gcode; my change allows to use any gcode to be injected. You can combine this.

Edit: I put an example to use "G27" in my branch's example configs.

File: Marlin/src/inc/SanityCheck.h [hunk 1]
```diff
@@ -551,6 +551,10 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
   #endif
 #endif
 
+#if defined(EVENT_GCODE_SD_STOP) && DISABLED(NOZZLE_PARK_FEATURE)
+  static_assert(NULL == strstr(EVENT_GCODE_SD_STOP, "M27"), "NOZZLE_PARK_FEATURE is required to use M27 in EVENT_GCODE_SD_STOP.");
```
Comment (robbycandra) [inline Marlin/src/inc/SanityCheck.h]:
```suggestion
  static_assert(NULL == strstr(EVENT_GCODE_SD_STOP, "G27"), "NOZZLE_PARK_FEATURE is required to use G27 in EVENT_GCODE_SD_STOP.");
```

**Comment by robbycandra** [comment]
@thinkyhead.  What about change

`#define EVENT_GCODE_SD_STOP "G28XY"
`
to 

`#define EVENT_GCODE_SD_STOP "G92.0 Z0\nG1 Z2\nG28 R0 X Y"`

At first i use homing, but i need z lift before moving x and y. And i found G27 is more simple.
But costumized GCODE at Stop Print is good.

**Comment by thinkyhead** [comment]
That's what the option is for. So you or anyone else can customize it to your needs. Basic default will be super simple.

**Comment by robbycandra** [comment]
> That's what the option is for. So you or anyone else can customize it to your needs. Basic default will be super simple.

Ah.. OK... I agree with you.

**Comment by prometx** [comment]
> @thinkyhead. What about change
> 
> `#define EVENT_GCODE_SD_STOP "G28XY" `
> to
> 
> `#define EVENT_GCODE_SD_STOP "G92.0 Z0\nG1 Z2\nG28 R0 X Y"`
> 
> At first i use homing, but i need z lift before moving x and y. And i found G27 is more simple.
> But costumized GCODE at Stop Print is good.

will this '\n' notation actually work for multiple commands to EVENT_GCODE_SD_STOP?

**Comment by qwewer0** [comment]
Yes, '\n' will work for multiple commands.

**Comment by prometx** [comment]
> Yes, '\n' will work for multiple commands.

Thank you very much! This will suit me I think.

Best,

**Pull request merged**